ui/components/MeshModelRegistry/MeshModelComponent.js

```diff
@@ -115,174 +114,168 @@ const MeshModelComponent = ({
-    }
-  };
+      if (response.data) {
+        const newData = searchText
+          ? [...response.data[view.toLowerCase()]]
+          : [...resourcesDetail, ...response.data[view.toLowerCase()]];
+        setResourcesDetail(newData);
 
-  // TODO: Use RTK
-  const getRelationships = async (page, sortOrder) => {
-    try {
-      const { total_count, relationships } = await getRelationshipsDetailWithPageSize(
-        page?.Relationships + 1,
-        rowsPerPage,
-        sortOrder.sort,
-        sortOrder.order,
-      );
-      setCount(total_count);
-      if (!isRequestCancelled && relationships) {
-        setResourcesDetail((prev) => [...prev, ...relationships]);
-        setSortOrder(sortOrder);
+        if (rowsPerPage !== 14) {
+          setRowsPerPage(14);
+        }
       }
-      setRowsPerPage(14);
     } catch (error) {
-      console.error('Failed to fetch relationships:', error);
+      console.error(`Failed to fetch ${view.toLowerCase()}:`, error);
     }
-  };
+  }, [
+    getMeshModelsData,
+    getComponentsData,
+    getRelationshipsData,
+    getRegistrantsData,
+    view,
+    searchText,
+    page,
+    rowsPerPage,
+    resourcesDetail,
+  ]);
 
-  // TODO: Use RTK
-  const getSearchedModels = async (searchText) => {
-    try {
-      const { total_count, models } = await searchModels(searchText, {
-        components: true,
-        relationships: true,
-      });
-      setCount(total_count);
+  const getRegistrants = async () => {
+    let registrantResponse;
+    let response;
+    registrantResponse = await getRegistrantsData(
+      {
+        params: {
+          page: searchText ? 1 : page.Registrants + 1,
+          pagesize: searchText ? 'all' : rowsPerPage,
+          search: searchText || '',
+        },
+      },
+      true,
+    );
+    if (registrantResponse.data && registrantResponse.data.registrants) {
+      const registrants = registrantResponse.data.registrants;
+      const tempResourcesDetail = [];
 
-      if (!isRequestCancelled) {
-        setResourcesDetail(models ? models : []);
+      for (let registrant of registrants) {
+        let hostname = toLower(registrant?.hostname);
+        const { data: modelRes } = await getMeshModelsData(
+          {
+            params: {
+              page: page?.Models + 1,
+              pagesize: rowsPerPage,
+              registrant: hostname,
+              components: true,
+              relationships: true,
+            },
+          },
+          true,
+        );
+
+        const updatedRegistrant = {
+          ...registrant,
+          models: modelRes.models || [],
+        };
+        tempResourcesDetail.push(updatedRegistrant);
       }
-    } catch (error) {
-      console.error('Failed to fetch components:', error);
+      response = {
+        data: {
+          registrants: tempResourcesDetail,
+        },
+      };
     }
+    setCount(response.total_count);
+    setRowsPerPage(14);
```

Comment:
make sure this random set of rows should not affect the handlescroll (sometime it could affect the scroll's div height)

Reply:
Yes it will not